developing-without-a-build/article.md

```diff
@@ -0,0 +1,266 @@
+const translationsPath = new URL(`./translations/${lang}.json`, import.meta.url);
+
+const response = await fetch(translationsPath);
+```
+
+[Read more about import.meta at MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import.meta)
+
+### Loading other packages
+When building an application you will quickly run into having to include other packages from npm. This works just fine in the browser, for example let's install and use lodash:
+
+```bash
+npm i --save lodash-es
+```
+
+```javascript
+import kebabCase from '../node_modules/lodash-es/kebabCase.js';
+
+console.log(kebabCase('camelCase'));
+```
+
+Lodash is a very modular library, the `kebabCase` function depends on a lot of other modules. These dependencies are taken care of automatically, the browser resolves and imports them for you:
+
+![](./images/lodash-es.png)
+
+Writing explicit paths to your node modules folder is a bit unusual. While it works just fine, most people are used to writing what's called a bare import specifier:
+```javascript
+import { kebabCase } from 'lodash-es';
+import kebabCase from 'lodash-es/kebabCase.js';
+```
+
+This doesn't work in the browser out of the box, as it doesn't know what file to request when it sees this path. Luckily there is a [new browser API called import maps](https://github.com/WICG/import-maps) which lets you instruct the browser how to resolve these imports:
+```html
+<script type="importmap">
+  {
+    "imports": {
+      "lodash-es": "./node_modules/lodash-es/lodash.js",
+      "lodash-es/": "./node_modules/lodash-es/"
+    }
+  }
+</script>
+```
+
+It's currently [implemented in chrome behind a flag](https://developers.google.com/web/updates/2019/03/kv-storage#import_maps), and it's easy to shim on other browsers [with es-module-shims](https://www.npmjs.com/package/es-module-shims). Until we get broad browser support, that can be an interesting option during development.
+
+If you don't want to go quite as cutting edge but still want to use the bare imports a great option is to use a web server which rewrites the bare imports on the fly into explicit paths before serving them to the browser. There are some servers available that do this. I recommend [es-dev-server](https://www.npmjs.com/package/es-dev-server), read more about it at the end of this article.
+
+### Caching
+Because we aren't bundling all of our code into just a few files, we don't have to set up any elaborate caching strategies. Your web server can use the file system's last modified timestamp to return a 304 if the file hasn't changed.
+
+You can test this in your browser by turning off `Disable cache` and refreshing:
+
+![](./images/caching.png)
+
+## Non-js modules
+So far we've only looked into javascript modules, and the story is looking pretty complete. It looks like we've got the tools needed to build different kinds of applications. But on the web we're not just writing javascript, we need to deal with other langauges as well.
```

Comment:
```suggestion
So far we've only looked into javascript modules, and the story is looking pretty complete. It looks like we've got the tools needed to build different kinds of applications. But on the web we're not just writing javascript, we need to deal with other languages as well.
```

Reply:
This got outdated, so I can't apply the changes anymore. Applied it manually, thanks!